perf(compiler): chain query creation instructions

crisbeto opened the pull request:
We always emit the query creation instructions in a group which makes them good candidates for chaining.

JoostK reviewed (commented):
The template pipeline has a dedicated phase for chaining:

https://github.com/angular/angular/blob/789f91bd4fd1ceca50b8cc765c0b7ccf5fc82768/packages/compiler/src/template/pipeline/src/phases/chaining.ts

The current approach may re-order queries (signal vs non-signal) which the chaining phase won't reorder.

crisbeto committed:
perf(compiler): chain query creation instructions

We always emit the query creation instructions in a group which makes them good candidates for chaining.

crisbeto commented on packages/compiler/src/render3/view/query_generation.ts:
```diff
@@ -177,15 +174,21 @@ export function createViewQueriesFunction(
+    } else {
+      viewQuerySignalCall ??= o.importExpr(R3.viewQuery);
+      viewQuerySignalCall = viewQuerySignalCall.callFn(params);
+    }
```
oops 😓
